### PR TITLE
Get DAO functions + pod burn rate

### PR DESCRIPTION
**Pull request by vadiminc**

### CODE REVIEW

File: contracts/SSVNetwork.sol [hunk 1]
```diff
@@ -632,16 +640,23 @@ contract SSVNetwork is OwnableUpgradeable, ISSVNetwork {
         emit ExecuteOperatorFeePeriodUpdate(newExecuteOperatorFeePeriod);
     }
 
+    function updateLiquidationThresholdPeriod(uint64 blocks) external onlyOwner override {
+        if(blocks < MINIMAL_LIQUIDATION_THRESHOLD) {
+            revert BelowMinimumBlockPeriod();
+        }
```
Comment (andrew-blox) [inline contracts/SSVNetwork.sol:646]: This is coming from V2 right?  
Is MINIMAL_LIQUIDATION_THRESHOLD = 50 also coming from V2, mainly the actual value of 50?

**Review by andrew-blox — CHANGES_REQUESTED** [review]

File: contracts/SSVNetwork.sol [hunk 2]
```diff
@@ -654,8 +669,10 @@ contract SSVNetwork is OwnableUpgradeable, ISSVNetwork {
         return (feeChangeRequest.fee.expand(), feeChangeRequest.approvalBeginTime, feeChangeRequest.approvalEndTime);
     }
 
-    function feeRecipientAddress(address recipientAddress) external override {
-        emit FeeRecipientAddressAdded(msg.sender, recipientAddress);
+    function getOperatorById(uint64 operatorId) external view override returns (address owner, uint256 fee, uint32 validatorCount) {
```
[the remark this answers is not in the document]
Comment (vadiminc) [reply contracts/SSVNetwork.sol:672]: @andrew-blox what the sense to have operatorSnaphot func then? Should I merge that logic into getOperatorById?

File: contracts/SSVNetwork.sol [hunk 3]
```diff
@@ -228,6 +232,10 @@ contract SSVNetwork is OwnableUpgradeable, ISSVNetwork {
         emit DeclaredOperatorFeeCancelation(msg.sender, operatorId);
     }
 
+    function feeRecipientAddress(address recipientAddress) external override {
+        emit FeeRecipientAddressAdded(msg.sender, recipientAddress);
+    }
+
```
[the remark this answers is not in the document]
Comment (vadiminc) [reply contracts/SSVNetwork.sol:238]: yes it is. Not sure why we see it here

File: contracts/SSVNetwork.sol [hunk 4]
```diff
@@ -632,16 +640,23 @@ contract SSVNetwork is OwnableUpgradeable, ISSVNetwork {
...
+        emit LiquidationThresholdPeriodUpdate(blocks);
+    }
+
     /************************************/
     /* Operator External View Functions */
     /************************************/
 
     function getOperatorFee(uint64 operatorId) external view override returns (uint256) {
-        Operator memory operator = _operators[operatorId];
+        if (_operators[operatorId].owner == address(0)) revert OperatorNotFound();
```
[the remark this answers is not in the document]
Comment (vadiminc) [reply contracts/SSVNetwork.sol:657]: no, we use as is - address as exists or not. And product requested do not restrict for deleted operators

File: contracts/SSVNetwork.sol [hunk 5]
```diff
@@ -669,13 +686,11 @@ contract SSVNetwork is OwnableUpgradeable, ISSVNetwork {
...
-        {
-            for (uint8 i = 0; i < operatorIds.length; ++i) {
-                Operator memory operator = _operators[operatorIds[i]];
-                if (operator.owner != address(0)) {
-                    podIndex += operator.snapshot.index + (uint64(block.number) - operator.snapshot.block) * operator.fee;
-                    burnRate += operator.fee;
-                }
+        for (uint8 i = 0; i < operatorIds.length; ++i) {
+            Operator memory operator = _operators[operatorIds[i]];
+            if (operator.owner != address(0)) {
```
[the remark this answers is not in the document]
Comment (vadiminc) [reply contracts/SSVNetwork.sol:691]: no

File: contracts/SSVNetwork.sol [hunk 6]
```diff
@@ -686,14 +701,25 @@ contract SSVNetwork is OwnableUpgradeable, ISSVNetwork {
...
         _validateHashedPod(owner, operatorIds, pod);
 
         return pod.disabled;
     }
 
+    function getPodBurnRate(uint64[] memory operatorIds) external view override returns (uint256) {
+        uint64 burnRate;
+        for (uint8 i = 0; i < operatorIds.length; ++i) {
+            Operator memory operator = _operators[operatorIds[i]];
+            if (operator.owner != address(0)) {
```
[the remark this answers is not in the document]
Comment (vadiminc) [reply contracts/SSVNetwork.sol:716]: no

File: contracts/SSVNetwork.sol (hunk 1 above)
Comment (vadiminc) [reply contracts/SSVNetwork.sol:646]: no, from initial test v3 logic

File: contracts/SSVNetwork.sol [hunk 7]
```diff
@@ -96,7 +96,7 @@ contract SSVNetwork is OwnableUpgradeable, ISSVNetwork {
     uint64 private _declareOperatorFeePeriod;
     uint64 private _executeOperatorFeePeriod;
     uint64 private _operatorMaxFeeIncrease;
-
+    uint64 private _minimumBlocksBeforeLiquidation;
```
Comment (mtabasco) [inline contracts/SSVNetwork.sol:99]: @vadiminc maybe you thought of this before but I'd rather ask. I see the purpose of this variable, and checking `_liquidatable` function that applies to pods, would it be possible to compute on the fly this `_minimumBlocksBeforeLiquidation` using operators' `_declareOperatorFeePeriod` and `_executeOperatorFeePeriod` ? I know they are measured in different units (blocks and timestamp). Just an idea. Thanks!

Comment (vadiminc) [reply contracts/SSVNetwork.sol:99]: @mtabasco thats ok, you can ask any questions ;) _minimumBlocksBeforeLiquidation, _declareOperatorFeePeriod and  _executeOperatorFeePeriod are not connected. _minimumBlocksBeforeLiquidation - says for how many bocks you should have tokens on your balance at least.  FeePeriod - about the operator fee changes flow.

File: contracts/SSVNetwork.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrew-blox) [reply contracts/SSVNetwork.sol:672]: @vadiminc yes you are right lets leave it out

File: contracts/SSVNetwork.sol (hunk 4 above)
[the remark this answers is not in the document]
Comment (andrew-blox) [reply contracts/SSVNetwork.sol:657]: ah yes im confusing these sorry.

**Review by andrew-blox — APPROVED** [review]

**Pull request merged**